src/Compilers/Core/MSBuildTaskTests/TestUtilities/DotNetSdkTestBase.cs

```diff
@@ -0,0 +1,194 @@
+
+            var dotnetInstallDir = Environment.GetEnvironmentVariable("DOTNET_INSTALL_DIR");
+            if (!IsMatchingDotNetInstance(s_dotnetInstallDir))
+            {
+                dotnetInstallDir = Environment.GetEnvironmentVariable("PATH").Split(Path.PathSeparator).FirstOrDefault(IsMatchingDotNetInstance);
+            }
+
+            if (dotnetInstallDir != null)
+            {
+                s_dotnetInstallDir = dotnetInstallDir;
+                s_dotnetSdkPath = GetSdkPath(s_dotnetInstallDir, s_dotnetSdkVersion);
+            }
+        }
+
+        private const string EmptyValueMarker = "--{empty}--";
+
+        private static void EmitTestHelperProps(
+            string objDirectory,
+            string projectFileName,
+            string projectXml)
+        {
+            // Common.props automatically import {project-name}.*.props files from MSBuildProjectExtensionsPath directory, 
+            // which is by defautl set to the IntermediateOutputPath:
+            File.WriteAllText(Path.Combine(objDirectory, projectFileName + ".TestHelpers.g.props"), projectXml);
+        }
+
+        private static string EmitTestHelperTargets(
+            string objDirectory, 
+            string outDirectory, 
+            string projectFileName, 
+            IEnumerable<string> expressions)
+        {
+            var outputFile = Path.Combine(outDirectory, "EvaluationResult.txt");
+
+            // Common.targets automatically import {project-name}.*.targets files from MSBuildProjectExtensionsPath directory, 
+            // which is by defautl set to the IntermediateOutputPath:
+            File.WriteAllText(Path.Combine(objDirectory, projectFileName + ".TestHelpers.g.targets"),
+$@"<Project>      
+  <Target Name=""Test_EvaluateExpressions"">
+    <PropertyGroup>
+      {string.Join(Environment.NewLine + "      ", expressions.SelectWithIndex((e, i) => $@"<_Value{i}>{e}</_Value{i}><_Value{i} Condition=""'$(_Value{i})' == ''"">{EmptyValueMarker}</_Value{i}>"))}
+    </PropertyGroup>
+    <ItemGroup>
+      <LinesToWrite Include=""{string.Join(";", expressions.SelectWithIndex((e, i) => $"$(_Value{i})"))}""/>
+    </ItemGroup>
+    <MakeDir Directories=""{outDirectory}"" />
+    <WriteLinesToFile File=""{outputFile}""
+                      Lines=""@(LinesToWrite)""
+                      Overwrite=""true""
+                      Encoding=""UTF-8"" />
+  </Target>
+
+  <!-- Overwrite CoreCompile target to avoid triggering the compiler -->
+  <Target Name=""CoreCompile""
+          DependsOnTargets=""$(CoreCompileDependsOn);_BeforeVBCSCoreCompile"">
+  </Target>
+
+  <Target Name=""InitializeSourceControlInformation""/>
+</Project>");
+
+            return outputFile;
+        }
+
+        public DotNetSdkTestBase()
+        {
+            DotNetPath = Path.Combine(s_dotnetInstallDir, s_dotnetExeName);
```

Comment:
>s_dotnetInstallDir [](start = 38, length = 18)

This will make an NRE if `s_dotnetInstallDir` is `null`. Perhaps we should throw a controlled exception at the end of the static ctor if we couldn't initialize `s_dotnetInstallDir`?

Reply:
The test will be skipped. 

Adding:
```
Assert.True(s_dotnetInstallDir != null, "SDK not found. Use ConditionalFact(typeof(DotNetSdkAvailable)) to skip the test if the SDK is not found.");
```